File: CountCombinations/comb.c
int BestOfSeries(int n) {
    char winner[1000];
    int  indx = 0;
    int  win_a = 0;
    int  win_b = 0;
    int  outcome = 0;
    
    bzero(winner, sizeof (winner));
    while (indx < n) {
        if (win_a >= series_win || win_b >= series_win) {
            outcome++;
            
            if (winner[indx] == 'a') {
                winner[indx] = 'b';
                win_a--;
                win_b++;
                indx++;
                continue;
            }
            
            if (winner[indx] == 'b') {
                while (indx >= 0 && winner[indx] == 'b') {
                    winner[indx--] = 0;
                    win_b--;
                }
                
                if (indx < 0) {
                    break;
                }
                
                if (winner[indx] == 'a') {
                    win_a--;
                    winner[indx] = 'b';
                    win_b++;
                }
            }
        }
        
        if (indx >= n) {
            indx = n - 1;
            while (indx >= 0 && winner[indx] == 'b') {
                winner[indx--] = 0;
                win_b--;
            }
            
            if (indx < 0) {
                break;
            }
            
            if (winner[indx] == 'a') {
                win_a--
            }
            
            winner[indx] = 'b';
            win_b++;
            continue;
        }
        switch (winner[indx]) {
            case 0:
                winner
            case 'a':
            
        }
    }
    return (0);
}

File: CountCombinations/comb.py
def BestOfSeries(n):
    winner = []
    for i in range(0, n):
        winner.append('')

    series_win = (n + 1)/2
    indx = 0
    win_a = 0
    win_b = 0
    possible = 0

    while indx >= 0:
        #print 'winner count ' , win_a, win_b
        if win_a >= series_win or win_b >= series_win:
            possible = possible + 1
            #print 'winner : ', indx, winner

            # if a is the last winner, then change the index to b
            if indx < n:
                if winner[indx] == 'a':
                    win_a = win_a - 1
                    winner[indx] = 'b'
                    win_b = win_b + 1
                    indx = indx + 1
                    continue

                if winner[indx] == 'b':
                    while indx >= 0 and winner[indx] == 'b':
                        winner[indx] = ''
                        indx = indx - 1
                        win_b = win_b - 1

                    if indx < 0:
                        break

                    if winner[indx] == 'a':
                        win_a = win_a - 1
                        winner[indx] = 'b'
                        win_b = win_b + 1
                    continue

        #print indx, winner

        if indx >=n:
            indx = n - 1
            while indx >= 0 and winner[indx] == 'b':
                winner[indx] = ''
                indx = indx - 1
                win_b = win_b - 1

            if indx < 0:
                break

            if winner[indx] == 'a':
                win_a = win_a - 1
            winner[indx] = 'b'
            win_b = win_b + 1
            continue

        if winner[indx] == '':
            winner[indx] = 'a'
            win_a = win_a + 1
        elif winner[indx] == 'a':
            if indx != n - 1:
                indx = indx + 1
                winner[indx] = 'a'
                win_a = win_a + 1
            else:
                winner[indx]= 'b'
                win_a = win_a - 1
                win_b = win_b + 1
        elif winner[indx] == 'b':
            indx = indx + 1
            if indx < n:
                winner[indx] = 'a'
                win_a = win_a + 1

    return possible

#for i in range(1,10):
#    print  i, BestOfSeries(i)
print BestOfSeries(15)
